### DISCUSSION TITLE
yq go library

### DISCUSSION THREAD

**Discussion by hobti01** [mikefarah/yq#1154] [General]
Thanks for the work, we use yq every day!

What are your thoughts on using yq as a go library?
- I notice the documentation section from a library perspective is a bit light :) https://pkg.go.dev/github.com/mikefarah/yq#section-documentation
- Would you consider the use of `yqlib` as a library appropriate or not in scope for what `yq` is designed for?

**Comment by mikefarah** [comment]
It's very light indeed!

It's not what I originally intended - and it would be neat if it was a separate lib - however I'm a little unsure of the extra admin time it will be for me to manage interfaces, versions, documentation and releases of two things. 

This has been asked before, and at the time the yqlib part of the code was in more of a flux so I didn't think it was quite ready. Now I think the interfaces are a more stable...just need time - or help?

**Reply by mikefarah** [reply]
In terms of help, and for context, this is my one and only go project. I'm not sure what the best practices are for go libraries. I'm not even sure how that page you linked to was generated :| but it's linked to the source code way back from v2.

**Comment by hobti01** [comment]
Thanks for sharing that background.  I am not a experienced go lib developer so I'm not the best to give advice.

we're looking to rewrite a collection of bash using yq into go and a library of yq would be logical to maintain the effort put into our evaluations.  However, I don't want to distract you from the core effort that many people rely upon.  Thanks for your timely response and your work on yq!
